src/kv/generic_serialise_wrapper.h

```diff
@@ -272,16 +281,24 @@ namespace kv
       return KvOperationType::KOT_NOT_SUPPORTED;
     }
 
+    // Should only be called once, once the GCM header and length of public
+    // domain have been read
+    void read_public_header()
+    {
+      is_snapshot = public_reader.template read_next<bool>();
```

Comment:
There is now a new snapshot flag in each serialised transaction (including write sets, where this flag is `false`).